genesyscloud/resource_genesyscloud_architect_schedules.go

```diff
@@ -0,0 +1,218 @@
+				Description: "Date time is represented as an ISO-8601 string without a timezone.",
+				Type:         schema.TypeString,
+				Required:     true,
+			},
+            "end": {
+				Description: "Date time is represented as an ISO-8601 string without a timezone.",
+				Type:         schema.TypeString,
+				Required:     true,
+			},
+            "rrule": {
+				Description: "An iCal Recurrence Rule (RRULE) string.",
+				Type:         schema.TypeString,
+				Required:     true,
+			},
+		},
+	}
+}
+
+func createArchitectSchedules(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	start := d.Get("start").(string)
+	end := d.Get("end").(string)
+	rrule := d.Get("rrule").(string)
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	archAPI := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	schedStart, err := time.Parse("2006-01-02T15:04:05.000000", start)
+	if err != nil {
+		return diag.Errorf("Failed to parse date %s: %s", start, err)
+	}
+
+	schedEnd, err := time.Parse("2006-01-02T15:04:05.000000", end)
+	if err != nil {
+		return diag.Errorf("Failed to parse date %s: %s", end, err)
+	}
+
+	sched := platformclientv2.Schedule{
+		Name:       		&name,
+		Description:      	&description,
+		Start:				&schedStart,
+		End:				&schedEnd,
+		Rrule:				&rrule,
+	}
+
+	log.Printf("Creating schedule %s", name)
+	schedule, _, getErr := archAPI.PostArchitectSchedules(sched)
+	if getErr != nil {
+		return diag.Errorf("Failed to create schedule %s: | Start: %s, | End: %s, | ERROR: %s", *sched.Name, *sched.Start, *sched.End, getErr)
+	}
+
+	d.SetId(*schedule.Id)
+
+	log.Printf("Created schedule %s %s", name, *schedule.Id)
+	return readArchitectSchedules(ctx, d, meta)
+}
+
+func readArchitectSchedules(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	archAPI := platformclientv2.NewArchitectApiWithConfig(sdkConfig)
+
+	log.Printf("Reading schedule %s", d.Id())
+
+	schedule, resp, getErr := archAPI.GetArchitectSchedule(d.Id())
+	if getErr != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Failed to read schedule %s: %s", d.Id(), getErr)
+	}
+
+	Start := new(string)
+	if schedule.Start != nil {
+		*Start = timeutil.Strftime(schedule.Start, "%Y-%m-%dT%H:%M:%S.%f")
+		
+	} else {
+		Start = nil
+	}
+	
+	End := new(string)
+	if schedule.End != nil {
+		*End = timeutil.Strftime(schedule.End, "%Y-%m-%dT%H:%M:%S.%f")
+		
+	} else {
+		End = nil
+	}
+
+	d.Set("name", *schedule.Name)
+	d.Set("description", *schedule.Description)
+	d.Set("start", Start)
+	d.Set("end", End)
+	d.Set("rrule", *schedule.Rrule)
+
+	log.Printf("Read schedule %s %s", d.Id(), *schedule.Name)
+	return nil
+}
+
+func updateArchitectSchedules(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
You should fetch the latest version of the resource when updating. This pattern has been applied in a few other resources: https://github.com/MyPureCloud/terraform-provider-genesyscloud/blob/main/genesyscloud/resource_genesyscloud_group.go#L235